pkg/trait/mount.go

```diff
@@ -165,4 +193,191 @@ func (t *mountTrait) mountResource(vols *[]corev1.Volume, mnts *[]corev1.VolumeM
 
 	*vols = append(*vols, *vol)
 	*mnts = append(*mnts, *mnt)
+
+	// User specified location file (only properties file)
+	if dstDir != "" {
+		if strings.HasSuffix(dstDir, ".properties") {
+			return []string{mntPath}, nil
+		}
+		return nil, nil
+	}
+
+	// We only process this for text configuration .properties files, never for resources
+	if conf.ContentType() == utilResource.ContentTypeText {
+		// the user asked to store the entire resource without specifying any filter
+		// we need to list all the resources belonging to the resource
+		if conf.StorageType() == utilResource.StorageTypeConfigmap {
+			cm := kubernetes.LookupConfigmap(e.Ctx, e.Client, e.Integration.Namespace, conf.Name())
```

Comment:
The operator must NOT access to the user's config map/secrets/resources.

Reply:
Isn't this already the case ? with mount.configs we know already what config maps/secrets are configuration resources.

Reply:
as far as I remember, it does not matter. 
The runtime/extensione loads them even if they are not .properties